File: test_register.js
const http = require('http');

const data = JSON.stringify({
    email: 'newtestuser2@example.com',
    username: 'newtestuser2',
    password: 'password123'
});

const options = {
    hostname: '127.0.0.1',
    port: 8000,
    path: '/api/auth/register/',
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        'Content-Length': data.length
    }
};

const req = http.request(options, (res) => {
    console.log(`STATUS: ${res.statusCode}`);
    res.setEncoding('utf8');
    res.on('data', (chunk) => {
        console.log(`BODY: ${chunk}`);
    });
});

req.on('error', (e) => {
    console.error(`problem with request: ${e.message}`);
});

req.write(data);
req.end();

File: test_register.py
import requests

url = 'http://127.0.0.1:8000/api/auth/register/'
data = {
    'email': 'newtestuser@example.com',
    'username': 'newtestuser',
    'password': 'password123'
}

response = requests.post(url, json=data)
print(f"Status Code: {response.status_code}")
print(f"Response: {response.text}")
